Client/src/view/BattleView/HandBox.java

```diff
@@ -307,15 +308,17 @@ void resetSelection() {
         if (player != null) {
             selectedCard = -1;
             updateCards();
-            updateNext();
         }
     }
 
     public void replaceSelectedCard() {
         if (selectedCard != -1) {
             String cardID = player.getHand().get(selectedCard).getCardId();
             battleScene.getController().replaceCard(cardID);
+            int currentTimesReplacedThisTurn = GameController.getInstance().getAvailableActions().getNumTimesReplacedThisTurn();
+            GameController.getInstance().getAvailableActions().setNumTimesReplacedThisTurn( currentTimesReplacedThisTurn + 1 );
```

Comment:
This is a minor point, but updating the number of times replaced should probably be the last line of code in this function. 

for example, what if the "updateNext()" function raises an error and does not actually replace the card?   In that case we then have 2 problems: firstly that the card didn't get replaced and 2nd that the number of replace this turn is incorrect.

Reply:
I believe its best if its not last, but if it happens right before we send updates to the client. I dont want to risk the chance of things being out of sync/order due to latency. Then again java does everything synchronously so this may not be a concern after all